contracts/0.4.24/Lido.sol

```diff
@@ -722,7 +590,171 @@ contract Lido is Versioned, StETHPermit, AragonApp {
         stakingRouter.deposit.value(depositsValue)(depositsCount, _stakingModuleId, _depositCalldata);
     }
 
-    /// DEPRECATED PUBLIC METHODS
+    /// @notice Mint shares backed by external vaults
+    ///
+    /// @param _receiver Address to receive the minted shares
+    /// @param _amountOfShares Amount of shares to mint
+    /// @return stethAmount The amount of stETH minted
+    ///
+    /// @dev authentication goes through isMinter in StETH
+    function mintExternalShares(address _receiver, uint256 _amountOfShares) external {
+        if (_receiver == address(0)) revert("MINT_RECEIVER_ZERO_ADDRESS");
+        if (_amountOfShares == 0) revert("MINT_ZERO_AMOUNT_OF_SHARES");
+
+        _whenNotStakingPaused();
+
+        uint256 stethAmount = super.getPooledEthByShares(_amountOfShares);
+
+        uint256 newExternalBalance = EXTERNAL_BALANCE_POSITION.getStorageUint256().add(stethAmount);
+        uint256 maxExternalBalance = _getMaxExternalBalance();
+
+        require(newExternalBalance <= maxExternalBalance, "EXTERNAL_BALANCE_LIMIT_EXCEEDED");
+
+        EXTERNAL_BALANCE_POSITION.setStorageUint256(newExternalBalance);
+
+        mintShares(_receiver, _amountOfShares);
+
+        emit ExternalSharesMinted(_receiver, _amountOfShares, stethAmount);
+    }
+
+    /// @notice Burns external shares from a specified account
+    ///
+    /// @param _amountOfShares Amount of shares to burn
+    ///
+    /// @dev authentication goes through _isBurner() method
+    function burnExternalShares(uint256 _amountOfShares) external {
+        if (_amountOfShares == 0) revert("BURN_ZERO_AMOUNT_OF_SHARES");
+
+        _whenNotStakingPaused();
+
+        uint256 stethAmount = super.getPooledEthByShares(_amountOfShares);
+        uint256 extBalance = EXTERNAL_BALANCE_POSITION.getStorageUint256();
+
+        if (extBalance < stethAmount) revert("EXT_BALANCE_TOO_SMALL");
+
+        EXTERNAL_BALANCE_POSITION.setStorageUint256(extBalance - stethAmount);
+
+        burnShares(msg.sender, _amountOfShares);
+
+        emit ExternalSharesBurned(msg.sender, _amountOfShares, stethAmount);
+    }
+
+    /// @notice processes CL related state changes as a part of the report processing
+    /// @dev all data validation was done by Accounting and OracleReportSanityChecker
+    /// @param _reportTimestamp timestamp of the report
+    /// @param _preClValidators number of validators in the previous CL state (for event compatibility)
+    /// @param _reportClValidators number of validators in the current CL state
+    /// @param _reportClBalance total balance of the current CL state
+    /// @param _postExternalBalance total balance of the external balance
```

Comment:
```suggestion
    /// @param _postExternalBalance total external ether balance
```

Reply:
Fixed in f560380d9679c50ac1677fe0a9d3b6e537bdf910